internal/mode/static/nginx/runtime/manager.go

```diff
@@ -116,3 +127,36 @@ func findMainProcess(
 
 	return pid, nil
 }
+
+func ensureNewNginxWorkers(
+	ctx context.Context,
+	childProcFile string,
+	previousContents []byte,
```

Comment:
The naming is a little confusing. You call it with "current", but in the function it's called "previous". Maybe call it "previous" in the caller as well.